### DISCUSSION TITLE
Multi Level Hierarchy Design

### DISCUSSION THREAD

**Discussion by snevarez1129** [FSMLang/FSMLang#117] [Q&A]
Is it possible to have a design where sub-machines have their own sub-machines? The design would have a top-level state machine with one sub-machine; the sub-machine would in turn have two of it's own sub-machines. Does FSMLang have any limitations for this kind of architecture?

**Comment by 9Fingered** [comment]
Hello, @snevarez1129 - Apologies for the slow answer!

Yes, sub machines can have sub-machines, though you'll need to have at least version 1.42; prior versions produced incorrect code for sub-sub machines.

Thank you for your interest in FSMLang!

**Comment by 9Fingered** [comment]
Hello, @snevarez1129 - please use version 1.44.1 if you are interested in your sub-sub-machines having data.
